### PR TITLE
AV-190960 Static Route Creation Fix

### PR DESCRIPTION
**Pull request by pkoshtavmware**
AV-190960 Static Route Creation Fix

### CODE REVIEW

**Comment by pkoshtavmware** [comment]
```
Running tool: /usr/local/go/bin/go test -timeout 30s -run ^TestMultipleBlockAffinityAddition$ github.com/vmware/load-balancer-and-ingress-services-for-kubernetes/tests/cnitests

ok  	github.com/vmware/load-balancer-and-ingress-services-for-kubernetes/tests/cnitests
```

**Comment by pkoshtavmware** [comment]
build ako

**Comment by pkoshtavmware** [comment]
Testing Done:
UnitTest Added:
```
Running tool: /usr/local/go/bin/go test -timeout 30s -run ^TestMultipleBlockAffinityAddition$ github.com/vmware/load-balancer-and-ingress-services-for-kubernetes/tests/cnitests

ok  	github.com/vmware/load-balancer-and-ingress-services-for-kubernetes/tests/cnitests
```
FT Added:
http://10.102.96.195/ft/test_report1.html

Other FT run results:
http://10.102.96.195/ft/test_report2.html
http://10.102.96.195/ft/test_report3.html
http://10.102.96.195/ft/test_report4.html

**Comment by pkoshtavmware** [comment]
build ako

**Comment by akshayhavile** [comment]
@pkoshtavmware Current UT pipeline failed in `static_route_test.go` so please check those UTs. It might be timing issue.

**Comment by pkoshtavmware** [comment]
build ako

**Comment by akshayhavile** [comment]
build ako

File: tests/cnitests/static_route_test.go [hunk 1]
```diff
@@ -545,3 +545,188 @@ func TestCiliumNodeAddDelete(t *testing.T) {
...
+	g.Expect(*nodes[0].StaticRoutes[4].RouteID).To(gomega.Equal("cluster-5"))
+	time.Sleep(10 * time.Second)
+
+	// deleting the BlockAffinity objects for the node
+	DynamicClient.Resource(lib.CalicoBlockaffinityGVR).Namespace("default").Delete(context.TODO(), nodeName1, v1.DeleteOptions{})
+	DynamicClient.Resource(lib.CalicoBlockaffinityGVR).Namespace("default").Delete(context.TODO(), "testblockaffinity2", v1.DeleteOptions{})
+	DynamicClient.Resource(lib.CalicoBlockaffinityGVR).Namespace("default").Delete(context.TODO(), nodeName2, v1.DeleteOptions{})
+	DynamicClient.Resource(lib.CalicoBlockaffinityGVR).Namespace("default").Delete(context.TODO(), "testblockaffinity4", v1.DeleteOptions{})
+	DynamicClient.Resource(lib.CalicoBlockaffinityGVR).Namespace("default").Delete(context.TODO(), "testblockaffinity5", v1.DeleteOptions{})
+	g.Eventually(func() int {
```
Comment (akshayhavile) [inline tests/cnitests/static_route_test.go:728]: can we check route id after deletion over here?

**Review by akshayhavile — DISMISSED** [review]
Overall changes looks good. one small comment

File: tests/cnitests/static_route_test.go [hunk 2]
```diff
@@ -545,3 +545,188 @@ func TestCiliumNodeAddDelete(t *testing.T) {
...
+	g.Expect(len(nodes[0].StaticRoutes)).To(gomega.Equal(5))
+	g.Expect(*nodes[0].StaticRoutes[0].RouteID).To(gomega.Equal("cluster-1"))
+	g.Expect(*nodes[0].StaticRoutes[1].RouteID).To(gomega.Equal("cluster-2"))
+	g.Expect(*nodes[0].StaticRoutes[2].RouteID).To(gomega.Equal("cluster-3"))
+	g.Expect(*nodes[0].StaticRoutes[3].RouteID).To(gomega.Equal("cluster-4"))
+	g.Expect(*nodes[0].StaticRoutes[4].RouteID).To(gomega.Equal("cluster-5"))
+	time.Sleep(10 * time.Second)
+
+	// deleting the BlockAffinity objects for the node
+	DynamicClient.Resource(lib.CalicoBlockaffinityGVR).Namespace("default").Delete(context.TODO(), nodeName1, v1.DeleteOptions{})
```
Comment (DixitAakash) [inline tests/cnitests/static_route_test.go:723]: Where are we creating the block affinities with names as the node names? I understand Calico will do it in a k8s setup but not sure where we have created it here.

**Review by DixitAakash — APPROVED** [review]
Overall changes look good to me.

File: tests/cnitests/static_route_test.go (hunk 2 above)
Comment (pkoshtavmware) [reply tests/cnitests/static_route_test.go:723]: Please check line 562, 577, 615 and 629

File: tests/cnitests/static_route_test.go (hunk 1 above)
Comment (pkoshtavmware) [reply tests/cnitests/static_route_test.go:725]: Done

**Comment by pkoshtavmware** [comment]
Also did manual testing with 3 node cluster, adding and deleting the intermediate node .Static routes got populated as expected also routesids were got reassigned after node deletion

**Comment by pkoshtavmware** [comment]
build ako

**Comment by pkoshtavmware** [comment]
build ako

**Review by DixitAakash — APPROVED** [review]
LGTM

**Review by akshayhavile — APPROVED** [review]

**Pull request merged**